Tasks/SonarQubePreBuild/SonarQubePreBuild.ps1

```diff
@@ -13,51 +13,34 @@ param(
 
 Write-Verbose "Starting SonarQube Pre-Build Setup Step"
 
-Write-Verbose -Verbose "connectedServiceName = $connectedServiceName"
-Write-Verbose -Verbose "projectKey = $projectKey"
-Write-Verbose -Verbose "projectName = $projectName"
-Write-Verbose -Verbose "cmdLineArgs = $cmdLineArgs"
-Write-Verbose -Verbose "configFile = $configFile"
-Write-Verbose -Verbose "dbConnectionString = $dbUrl"
-Write-Verbose -Verbose "dbUsername = $dbUsername"
+Write-Verbose "connectedServiceName = $connectedServiceName"
+Write-Verbose "projectKey = $projectKey"
+Write-Verbose "projectName = $projectName"
+Write-Verbose "cmdLineArgs = $cmdLineArgs"
+Write-Verbose "configFile = $configFile"
+Write-Verbose "dbConnectionString = $dbUrl"
 
-. ./SonarQubeHelper.ps1
-
-$prcaEnabled = Get-TaskVariable -Context $distributedTaskContext -Name "PullRequestSonarQubeCodeAnalysisEnabled"
-if ($prcaEnabled -ieq "true")
-{
-    if ($cmdLineArgs -and $cmdLineArgs.ToString().Contains("sonar.analysis.mode"))
-    {
-        throw "Error: sonar.analysis.mode seems to be set already. Please check the properties of SonarQube build tasks and try again."
-    }
-
-    Write-Verbose -Verbose "PullRequestSonarQubeCodeAnalysisEnabled is true, setting command line args for incremental mode for sonar-runner..."
+import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
+. ./SonarQubePreBuildImpl.ps1
 
-    $cmdLineArgs = $cmdLineArgs + " " + "/d:sonar.analysis.mode=incremental"
 
-    #use this variable in post-test task
-    SetTaskContextVariable "SonarqubeAnalysisModeIsIncremental" "true"
-}
-
-import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
+$cmdLineArgs = UpdateArgsForPullRequestAnalysis $cmdLineArgs
+Write-Verbose -Verbose $cmdLineArgs
```

Comment:
using -verbose option here? For consistency, it's better to remove them from all log Write-Verbose statements and use Write-Host where we need to always print..